osu.Game/Screens/SelectV2/RealmPopulatingOnlineLookupSource.cs

```diff
@@ -42,7 +42,14 @@ public partial class RealmPopulatingOnlineLookupSource : Component
             var request = new GetBeatmapSetRequest(id);
             var tcs = new TaskCompletionSource<APIBeatmapSet?>();
 
-            request.Success += onlineBeatmapSet =>
+            // async request success callback is a bit of a dangerous game, but there's some reasoning for it.
+            // - don't really want to use `IAPIAccess.PerformAsync()` because we still want to respect request queueing & online status checks
+            // - we want the realm write here to be async because it is known to be slow for some users with large beatmap collections
+            // - at the time of writing `RealmAccess.WriteAsync()` can only be safely called from update thread,
+            //   and API request completion callbacks are automatically marshaled onto update thread scheduler,
+            //   so calling `WriteAsync()` within the callback is a somewhat "nice" way of guaranteeing that the call is safe
+            //   (rather than having to enforce that `GetBeatmapSetAsync()` can only be called from update thread, or locally scheduling)
+            request.Success += async onlineBeatmapSet =>
```

Comment:
i'm surprised inspections allow this to exist (fire-and-force async void assigned to non-async method), but if it works let's go with it 😅